lib/apiservers/engine/backends/cache/layer_cache.go

```diff
@@ -15,18 +15,27 @@
 package cache
 
 import (
+	"encoding/json"
+	"fmt"
 	"sync"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/vic/lib/apiservers/engine/backends/kv"
+	"github.com/vmware/vic/lib/apiservers/portlayer/client"
+	"github.com/vmware/vic/lib/metadata"
+	"github.com/vmware/vic/pkg/trace"
 )
 
 // LCache is an in-memory cache to account for existing image layers
 // It is used primarily by imagec when coordinating layer downloads
 // The cache is initially hydrated by way of the image cache at startup
 type LCache struct {
 	m sync.RWMutex
-	// layers maps from layer ID to boolean
-	// true indicates that the layer is currently downloading
-	// false indicates that the layer has been downloaded and written to the portlayer
-	layers map[string]bool
+	// Layers maps from layer id to layer metadata
+	Layers map[string]*metadata.LayerConfig
```

Comment:
I'm actually going to ditch the metadata.LayerConfig and replace it with the ImageWithMeta struct we use in imagec.